### PR TITLE
integration/cri-o: add integration tests kata 2.x ready

### PR DESCRIPTION
**Pull request by fgiudici**
Add back integration tests with CRI-O but 2.x ready as per issue https://github.com/kata-containers/tests/issues/3592.
Once kata is installed on the system, "make crio" should allow the tests to be run.

### CODE REVIEW

**Comment by fgiudici** [comment]
CC: @fidencio , @littlejawa

File: Makefile [hunk 1]
```diff
@@ -99,6 +99,10 @@ else
 	bash sanity/check_sanity.sh
 endif
 
+crio:
+	bash .ci/install_bats.sh
+	RUNTIME=${RUNTIME} ./integration/cri-o/cri-o.sh
```
Comment (fidencio) [inline Makefile:104]: Which versions of CRI-O we care about testing?
RUNTIME doesn't work with newer versions, those use CONTAINER_DEFAULT_RUNTIME, IIRC.
CONTAINER_DEFAULT_RUNTIME doesn't work with older versions, though.

So, the best way to go for now is exporting both.

File: integration/cri-o/cri-o.sh [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+# Skip CRI-O tests that currently are not working
+pushd "${crio_repository_path}/test/"
+for i in "${skipCRIOTests[@]}"
+do
+	sed -i '/'${i}'/a skip \"This is not working\"' "$GOPATH/src/${crio_repository}/test/ctr.bats"
+done
+
+IFS=$OLD_IFS
+
+# On other distros or on ZUUL, use overlay.
```
Comment (fidencio) [inline integration/cri-o/cri-o.sh:74]: ZUUL is no more a thing we use. :-)

Anyways, we should always use overlay with CRI-O, as devmapper is not exactly maintained there.

File: integration/cri-o/cri-o.sh [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+# This will allow us to run tests with at least 2 different
+# storage drivers.
+export STORAGE_OPTIONS="$OVERLAY_STORAGE_OPTIONS"
```
Comment (fidencio) [inline integration/cri-o/cri-o.sh:77]: I see the point of doing this, but if unless we get it to fully work and being fully maintained on CRI-O side, I wouldn't bother testing with device-mapper there.

**Review by fidencio — COMMENTED** [review]
@fgiudici, thanks for working on this, I've added a few comments as a review.

In general, solid work.

**Comment by fidencio** [comment]
And the static checks are failing due to:
```
 ERROR: Commit 4f783e3bed9a5fcd80bdae2da69e55512635a22c: Failed to find subsystem in subject: "Revert \"integration: Remove old CRI-O tests\""
ERROR: checkcommits failed. See the document below for help on formatting
commits for the project.

https://github.com/kata-containers/community/blob/master/CONTRIBUTING.md#patch-format
```

File: Makefile (hunk 1 above)
Comment (fgiudici) [reply Makefile:104]: Uh, missed it!  I moved the env vars directly in the cri-o.sh script, RUNTIME here was ignored, so removing it!
The vars have shimv2 compatible defaults that will be picked up for CRI-O configuration by the CRI-O test script. The user may still export its own values if needed (exporting CONTAINER_DEFAULT_RUNTIME, RUNTIME_ROOT and RUNTIME_TYPE) but I would leave them out of the Makefile... I don't expect a use case where the user needs to rediefine the vars too. Could this option be viable?


> And the static checks are failing due to:
> 
> ```
> ERROR: Commit 4f783e3bed9a5fcd80bdae2da69e55512635a22c: Failed to find subsystem in subject: "Revert \"integration: Remove old CRI-O tests\""
> ERROR: checkcommits failed. See the document below for help on formatting
> commits for the project.
> 
> https://github.com/kata-containers/community/blob/master/CONTRIBUTING.md#patch-format
> ```

File: integration/cri-o/cri-o.sh (hunk 2 above)
Comment (fgiudici) [reply integration/cri-o/cri-o.sh:74]: Oh, sure, removed :-)

File: integration/cri-o/cri-o.sh (hunk 3 above)
Comment (fgiudici) [reply integration/cri-o/cri-o.sh:77]: Ok, I left the variable exported with the driver explicitly set to "overlay". This should be the default in CRI-O, so probably not needed...  but leaving it explicit doesn't harm, moved near the exported vars for CRI-O test script.

**Comment by fgiudici** [comment]
@fidencio , many thanks for the review (and the hint for resolving the static check issue)!
Enforced also build of CRI-O binaries: we may download the repo or just change the branch, so let's ensure we also (re)build the binaries to match the expected CRI-O version.

**Review by jodh-intel — APPROVED** [review]
Thanks @fgiudici.

lgtm

**Review by fidencio — APPROVED** [review]
lgtm, thanks @fgiudici!

**Pull request merged**